### ISSUE TITLE
[1.10.2 SMP] Clicking Edit in Quest Line GUI Crashes Client

### ISSUE THREAD

**Issue by ArmandVestibule** [Funwayguy/BetterQuesting#113]
Summary:

Following an upgrade to the latest version of BQ and Standard Expansion (with NPC expansion installed as well), clicking Edit inside the Quest Line view causes an immediate client crash. I have cleared out all quest jsons to be sure it wasn't a bad file in that area.

The summary of the crash is as follows:

```
java.lang.NullPointerException: Updating screen events
	at betterquesting.client.gui.GuiQuestLinesMain.func_73864_a(GuiQuestLinesMain.java:178)
	at net.minecraft.client.gui.GuiScreen.func_146274_d(GuiScreen.java:548)
	at betterquesting.api.client.gui.GuiScreenThemed.func_146274_d(GuiScreenThemed.java:145)
	at net.minecraft.client.gui.GuiScreen.func_146269_k(GuiScreen.java:517)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1700)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1055)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
```

Full Crash Log: [crash-2016-12-29_02.06.04-client.txt](https://github.com/Funwayguy/BetterQuesting/files/676749/crash-2016-12-29_02.06.04-client.txt)

As per these logs, I am running the base mod at version 2.3.193, NPC Integration at 2.3.19, and the Standard Expansion at 2.3.118.

I had made this upgrade with hopes that similar issues as seen here may have been resolved, such as Hunt tasks resetting to "kill one zombie" in all quests as well as Issue #77  .

Thanks and regards!

**Comment by Funwayguy** [comment]
I made a really dumb typo it seems. Fix coming out shortly, just fixing up some other issues with party invites first

**Comment by ArmandVestibule** [comment]
Sounds great. I'll grab the updates as soon as they become available and report back.

Thanks for your effort!

**Comment by Funwayguy** [comment]
The fix has been sent off to Curse for approval. Should be downloadable soon

**Comment by ArmandVestibule** [comment]
Okay. Great progress on this, however I feel there may have been a new issue introduced.

While I can, now, get into the editor and manipulate quests I cannot view quest details from the player's perspective.  In other words, if I open the quest UI, select a quest line and then click one of the unlocked, available quests, nothing happens.

This is consistent in both edit mode and normal mode. The advanced info tooltip continues to function and it does list the number of tasks and their completion.

Lastly, as part of troubleshooting, I invoked /bq_admin reset all, which had no effect other than resetting all of my mob hunt quests back to "hunt one zombie" and removed all rewards. Is this by design?

In summary:
1) Initial problem is resolved.
2) Can no longer view quest details in the quest line tree as a player or OP.
3) bq_admin reset all wipes out all quest objectives and rewards and defaults Mob Hunt quests to kill one zombie. 

I can open new issues, if you would prefer, for tracking purposes.

Thanks!

**Comment by Funwayguy** [comment]
1) Yay \o/

2) Now fixed in dev for next update

3) Working on this now

**Comment by ArmandVestibule** [comment]
Awesome. Thanks!

**Comment by Funwayguy** [comment]
I've sent out a fix for 2 but until I can recreate 3 there isn't much I can do about it. You can open up a new issue for it if you like

**Comment by ArmandVestibule** [comment]
2 is fixed for me now -- thanks!

I cannot remove tasks or rewards from a quest once they've been added, however. The X does nothing.

As for the last item, yeah -- I'll poke around. I'm using a rather heavily modded setup and it's nothing that regular backups and restorations can't fix while editing. So I should have a workaround in play for that.

**Comment by Funwayguy** [comment]
Pretty sure I fixed 3 as of a recent update when I discovered I did some dumb stuff with syncing (or to be more specific a lack of syncing altogether) so I'll close this thread now
